src/app/src/main/kotlin/org/imperial/mrc/hint/Extensions.kt

```diff
@@ -51,16 +53,22 @@ fun Response.asResponseEntity(): ResponseEntity<String> {
     return try {
         val body = this.body().asString("application/json")
         val json = ObjectMapper().readTree(body)
-        if (!json.has("status")) {
+        if (!json.has("status") && !json.has("result")) {
             throw IOException()
         }
-
-        ResponseEntity.status(httpStatus)
-                .contentType(MediaType.APPLICATION_JSON)
-                .body(body)
+        return if (json.has("status")) {
+            // this is a hintr response, so already conforms to our response schema
+            ResponseEntity.status(httpStatus)
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .body(body)
+        } else {
+            // this is an ADR response, so convert to our response schema
+            SuccessResponse(json["result"])
+                    .asResponseEntity()
+        }
 
     } catch (e: IOException) {
-        ErrorDetail(httpStatus, "Could not parse response.")
+        ErrorDetail(httpStatusFromCode(500), "Could not parse response.")
```

Comment:
Not related to this changeset but if we're returning an error response, we always want this to be a 500, even if the original request returned a different status code. See updated test of this here: https://github.com/mrc-ide/hint/pull/273/files#diff-5db22b6b5a666b3836c7f617a4378d0eR76